controllers/statusmanager/status_manager.go

```diff
@@ -5,64 +5,315 @@ package statusmanager
+	status := StatusManager{
+		client:            client,
+		mapper:            mapper,
+		name:              name,
+		version:           version,
+		OperatorNamespace: operatorNamespace,
+		AdaptorName:       sharedInfo.AntreaPlatform,
+	}
+	if sharedInfo.AntreaPlatform == "openshift" {
+		status.Adaptor = &StatusOc{}
+	} else {
+		status.Adaptor = &StatusK8s{}
+	}
+	return &status
+}
+
+func (status *StatusManager) setConditions(progressing []string, reachedAvailableLevel bool) {
+	conditions := make([]configv1.ClusterOperatorStatusCondition, 0, 2)
+	if len(progressing) > 0 {
+		conditions = append(conditions,
+			configv1.ClusterOperatorStatusCondition{
+				Type:    configv1.OperatorProgressing,
+				Status:  configv1.ConditionTrue,
+				Reason:  "Deploying",
+				Message: strings.Join(progressing, "\n"),
+			},
+		)
+	} else {
+		conditions = append(conditions,
+			configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorProgressing,
+				Status: configv1.ConditionFalse,
+			},
+		)
+	}
+	if reachedAvailableLevel {
+		conditions = append(conditions,
+			configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorAvailable,
+				Status: configv1.ConditionTrue,
+			},
+		)
+	}
+	status.set(status, reachedAvailableLevel, conditions...)
+}
+
+// Set updates the AntreaInstall.Status with the provided conditions for platform kubernetes.
+func (adaptor *StatusK8s) set(status *StatusManager, reachedAvailableLevel bool, conditions ...configv1.ClusterOperatorStatusCondition) {
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		antreaInstall := &operatorv1.AntreaInstall{}
+		err := status.client.Get(context.TODO(), types.NamespacedName{Namespace: operatortypes.OperatorNameSpace, Name: operatortypes.OperatorConfigName}, antreaInstall)
+		if err != nil {
+			log.Error(err, "Failed to get antreaInstall")
+			return err
+		}
+		co := &configv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: status.name}}
+		oldStatus := antreaInstall.Status.DeepCopy()
+		if reachedAvailableLevel {
+			co.Status.Versions = []configv1.OperandVersion{
+				{Name: "operator", Version: version.Version},
+			}
+		}
+		status.CombineConditions(&co.Status.Conditions, &conditions)
+		progressingCondition := v1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorProgressing)
+		availableCondition := v1helpers.FindStatusCondition(co.Status.Conditions, configv1.OperatorAvailable)
+		if availableCondition == nil && progressingCondition != nil && progressingCondition.Status == configv1.ConditionTrue {
+			v1helpers.SetStatusCondition(&co.Status.Conditions,
+				configv1.ClusterOperatorStatusCondition{
+					Type:    configv1.OperatorAvailable,
+					Status:  configv1.ConditionFalse,
+					Reason:  "Startup",
+					Message: "The network is starting up",
+				},
+			)
+		}
+		v1helpers.SetStatusCondition(&co.Status.Conditions,
```

Comment:
innocent question: does this function return an error?

Reply:
no, it doesn't return an error.